### ISSUE TITLE
ContextManager.ActiveSpan is null in ApplicationOnEndRequest

### ISSUE THREAD

**Issue by DevinShaw** [OpenSkywalking/skywalking-netcore#131]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

### Bug
- Which version of SkyWalking, OS and .NET Runtime?
SkyWalking: 0.7.0
OS: win7
.NET Runtime: 4.6.1 

- Which  project?
SkyWalking.Sample.AspNet

- What happen?
When I add **async action** to ValuesController in SkyWalking.Sample.AspNet, I find that ContextManager.ActiveSpan is always null.  

```
    public void ApplicationOnEndRequest(object sender, EventArgs e)
    {
        var httpApplication = sender as HttpApplication;
        var httpContext = httpApplication.Context;
        if (httpContext.Request.HttpMethod == "OPTIONS")
        {
            //asp.net Exclude OPTIONS request
            return;
        }
        var httpRequestSpan = ContextManager.ActiveSpan ;
        if (httpRequestSpan == null)
        {
            return;
        }
    }

```

**Comment by liuhaoyang** [comment]
This is due to the AsyncLocal cross-threading bug in the .NET Framework.

**Comment by liuhaoyang** [comment]
Repeat with https://github.com/OpenSkywalking/skywalking-netcore/issues/102, this bug will be fixed in V0.8.
